gcloud/bigquery/job.py

```diff
@@ -0,0 +1,702 @@
+        :returns: The client passed in or the currently bound client.
+        """
+        if client is None:
+            client = self._client
+        return client
+
+    def _populate_config_resource(self, configuration):
+
+        if self.allow_jagged_rows is not None:
+            configuration['allowJaggedRows'] = self.allow_jagged_rows
+        if self.allow_quoted_newlines is not None:
+            configuration['allowQuotedNewlines'] = self.allow_quoted_newlines
+        if self.create_disposition is not None:
+            configuration['createDisposition'] = self.create_disposition
+        if self.encoding is not None:
+            configuration['encoding'] = self.encoding
+        if self.field_delimiter is not None:
+            configuration['fieldDelimiter'] = self.field_delimiter
+        if self.ignore_unknown_values is not None:
+            configuration['ignoreUnknownValues'] = self.ignore_unknown_values
+        if self.max_bad_records is not None:
+            configuration['maxBadRecords'] = self.max_bad_records
+        if self.quote_character is not None:
+            configuration['quote'] = self.quote_character
+        if self.skip_leading_rows is not None:
+            configuration['skipLeadingRows'] = self.skip_leading_rows
+        if self.source_format is not None:
+            configuration['sourceFormat'] = self.source_format
+        if self.write_disposition is not None:
+            configuration['writeDisposition'] = self.write_disposition
+
+    def _build_resource(self):
+        """Generate a resource for ``begin``."""
+        resource = {
+            'jobReference': {
+                'projectId': self.project,
+                'jobId': self.name,
+            },
+            'configuration': {
+                'load': {
+                    'sourceUris': self.source_uris,
+                    'destinationTable': {
+                        'projectId': self.destination.project,
+                        'datasetId': self.destination.dataset_name,
+                        'tableId': self.destination.name,
+                    },
+                },
+            },
+        }
+        configuration = resource['configuration']['load']
+        self._populate_config_resource(configuration)
+
+        if len(self.schema) > 0:
+            configuration['schema'] = {
+                'fields': _build_schema_resource(self.schema)}
+
+        return resource
+
+    def _set_properties(self, api_response):
+        """Update properties from resource in body of ``api_response``
+
+        :type api_response: httplib2.Response
+        :param api_response: response returned from an API call
+        """
+        self._properties.clear()
+        cleaned = api_response.copy()
+        schema = cleaned.pop('schema', {'fields': ()})
+        self.schema = _parse_schema_resource(schema)
+
+        statistics = cleaned.get('statistics', {})
+        if 'creationTime' in statistics:
```

Comment:
Well for one, if the property is never accessed it's a tiny performance gain. For two, it delegates the need for conversion to the methods already doing checking / conversion. Here, it's a bit out of place. (Honestly, having autogenerated code based on a schema in a discovery doc would make us "happy".)